app/javascript/mastodon/components/scrollable_list.js

```diff
@@ -95,7 +96,12 @@ export default class ScrollableList extends PureComponent {
 
   getFirstChildKey (props) {
     const { children } = props;
-    const firstChild = Array.isArray(children) ? children[0] : children;
+    let firstChild = children;
+    if (children instanceof ImmutableList) {
+      firstChild = children.get(0);
+    } else if (Array.isArray(children)) {
```

Comment:
It seems that ScrollableList is always got children as Immutable.List. So, this code perhaps isn't needed.